src/control/server/storage/config.go

```diff
@@ -313,3 +298,11 @@ func (bc *BdevConfig) Validate(class Class) error {
 
 	return nil
 }
+
+type StorageConfig struct {
+	Tiers      Configs `yaml:"storage"`
+	TiersNum   int     `yaml:"-"` // @todo_llasek: cmdLongFlag:"--nvme_tiers" cmdShortFlag:"-T"`
+	ConfigPath string  `yaml:"-" cmdLongFlag:"--nvme" cmdShortFlag:"-n"`
+	MemSize    int     `yaml:"-" cmdLongFlag:"--mem_size,nonzero" cmdShortFlag:"-r,nonzero"`
+	VosEnv     string  `yaml:"-" cmdEnv:"VOS_BDEV_CLASS"`
+}
```

Comment:
I don't understand this change. Why do we now have `storage.Config` and `storage.StorageConfig`? The idea behind my changes was that a given engine would have N `storage.Config` entries, each representing a single tier.

Reply:
These are global arguments for the engine.
Note the engine (or rather SPDK) does take a single config file describing all devices.
The VOS env is global anwyay. IMO that can be completely removed but trying to limit the necessary changes for 2.0.
The MemSize is for in-memory blobs. This isn't a production feature, nor did it even work a couple of weeks ago. Leaving as a global value, applicable to all tiers.

Reply:
OK, I see where you're going with this now. Please take a look at this PR which should merge cleanly on top of your current PR to improve some naming/clarity: https://github.com/daos-stack/daos/pull/5781